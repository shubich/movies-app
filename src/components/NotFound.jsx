import React from 'react';
import {Film} from './Film';
import { movies } from '../data';

export const NotFound = () => (
    <div id='notfound'>
        <div className='container'>
            <h2>No films found</h2>
        </div>
    </div>
);