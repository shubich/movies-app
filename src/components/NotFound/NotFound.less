@import "../../styles/palette";
@import "../../styles/spaces";
@import "../../styles/media";

.NotFound {
    padding: @desktop-space;
    background: white;
    min-height: 550px;
    display: flex;
    align-items: center;
    @media @phone {
        padding: @phone-space;
        min-height: 50vh;
    }  
}

.NotFound-Header {
    margin: unset;
    text-align: center;
    color: @basic-color;
    font-size: 3em;
    font-weight: unset;    
    @media @phone {
        font-size: 2em;
    }  
}

.NotFound-Query {
    b {
        color: #ccc;
        font-weight: 100;
    }
    &::before {
        content: ' ';
    }
}
