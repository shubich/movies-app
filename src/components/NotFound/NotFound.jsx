import React from 'react';
import PropTypes from 'prop-types';

import './NotFound.less';

const NotFound = ({ query }) => (
  <div className="NotFound">
    <div className="Container">
      <h2 className="NotFound-Header">
        No results found
        {
          query ? (
            <span className="NotFound-Query">
              for <b>&quot;{query}&quot;</b>
            </span>
          ) : (null)
        }
      </h2>
    </div>
  </div>
);

NotFound.propTypes = {
  query: PropTypes.string,
};

NotFound.defaultProps = {
  query: '',
};

export default NotFound;
