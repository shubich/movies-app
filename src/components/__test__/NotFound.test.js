import React from 'react';
import { NotFound } from '../NotFound';
import { shallow } from 'enzyme';

describe('NotFound', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <NotFound />
        );
        expect(wrapper).toMatchSnapshot();
    });
})