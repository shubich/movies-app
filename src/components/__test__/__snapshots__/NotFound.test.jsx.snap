// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <h2>
          No films found
        </h2>
      </div>,
      "id": "notfound",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          No films found
        </h2>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "No films found",
        },
        "ref": null,
        "rendered": "No films found",
        "type": "h2",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <h2>
            No films found
          </h2>
        </div>,
        "id": "notfound",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            No films found
          </h2>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "No films found",
          },
          "ref": null,
          "rendered": "No films found",
          "type": "h2",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
